### DISCUSSION TITLE
SupabaseClient.ts: Uncaught Error: supabaseUrl is required.

### DISCUSSION THREAD

**Discussion by bibinbiju111** [supabase/supabase#14631] [Questions]
I have  installed supabase and provided the url and api key in the .env (doing a react project) folder. I correctly imported it in the following way:

import { createClient } from "@supabase/supabase-js";
const supabase = createClient(supabaseURL,supabaseKEY);
export default supabase;    where the parameters are my variable names.

I actually wanted a  successful login implementation when I click the login button with the email and password it should verify it with the table in the database  called  User where I provided it. I dont' need a new user registeration or a signup , just a basic login function. And if the user entered credentials is equal to that of the ones in the User table then it should be redirected to another page (/homePage).

When I ran npm start in the terminal, it compiles with 0 error and localhost is white screen. In the inspect option and select the console those errors are mentioned. 
Please provide a good help. Looking forward for the fast reply.Following is each component code :

-------client.js--------

import { createClient } from "@supabase/supabase-js";
const supabaseURL=process.env.SUPABASE_URL
const supabaseKEY = process.env.SUPABASE_API_KEY;
// Initialize Supabase client
const supabase = createClient(supabaseURL,supabaseKEY);
export default supabase;

--------fetchUserCredentials.js-----------

import supabase from "../config/client";
// Fetch user credentials from Supabase table
const fetchUserCredentials = async () => {
  const { data, error } = await supabase.from("User").select("*");
  if (error) {
    console.error("Error fetching user credentials:", error);
  }
  return data;
};
export default fetchUserCredentials;

-------logintrial.js--------

import React,{ useState } from "react";
import { useNavigate } from "react-router-dom";
import fetchUserCredentials from "./fetchUserCredentials";

const Logintrial = () => {
  const navigate = useNavigate();
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const handleLogin = async () => {
    const data = await fetchUserCredentials();
    const matchedUser = data.find(
      (user) => user.email === username && user.password === password
    );

    if (matchedUser) {
      // Perform any necessary login actions (e.g., set authentication token, etc.)
      // Redirect the user to the userHome page
      navigate("/UserHome");
    } else {
      console.log("Invalid credentials. Please try again.");
    }
  };

  return (
    <div>
      <input
        type="text"
        placeholder="Username"
        value={username}
        onChange={(e) => setUsername(e.target.value)}
      />
      <input
        type="password"
        placeholder="Password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <button onClick={handleLogin}>Login</button>
    </div>
  );
};

export default Logintrial;

**Comment by GaryAustin1** [comment]
Do:
```
const supabaseURL=process.env.SUPABASE_URL
console.log(supabaseURL)
```
And see if you have a URL.   Seems your env setup is not working.

**Reply by bibinbiju111** [reply]
That was spot on .Thanks Sir. I have changed it and used the url and key directly to the variable. Now the site loads. But when I am in the login page and submit the same values in the table it says invalid credentials from logintrial.js , which I pasted above. I think something is wrong about the if loop and 'matchedUser' thing in the loginpage.js or is it something with variable name? Kindly do check sir?

**Reply by GaryAustin1** [reply]
You probably should consider using Supabase Discord for help on debugging your code.  It is more interactive and your issue likely depends on the environment you are running in.  It is very unclear what User is and how you have credentials to signin available before you signin...

**Reply by bibinbiju111** [reply]
Ok sir, fine I will surely join now. Hope meet there soon. So how shall I provide the codes up there?

**Reply by GaryAustin1** [reply]
You use tripple ``` there surrounding your code.

**Reply by bibinbiju111** [reply]
I have put it there in #help and questions server.Thanks
